### DISCUSSION TITLE
add USD Prims to USD collection cause maya freeze

### DISCUSSION THREAD

**Discussion by sylvieChen** [Autodesk/maya-usd#3854] [Q&A]
Hi,
I would like to ask for some help. I am trying to add usd prims to usd collections using usd api. I find when I do it on a heavy  usd assembly, even when I am adding only one prim to a already exists collection it still cause maya to freeze. I tried to change the script to use Sdf API, and put it under Sdf.ChangeBlock(), although the script runs faster, but after execute the script, maya still freeze for about 1-2 minutes.
```
    collection = Usd.CollectionAPI.Apply(root_prim, set_name)
    collection.CreateExpansionRuleAttr(Usd.Tokens.explicitOnly)
    collection.CreateIncludesRel().AddTarget(prim.GetPath())
```
I tried to use cmds.refresh(suspend=True) before running the script to suspend maya viewport to redraw, but maya still freeze. 
if you have any idea, please let me know. 

I am testing on maya 2023, mayausd 0.26.0

**Comment by wallworm** [comment]
Thank you for reporting. We are currently exploring collection support in our UI for light linking and generic collection use. We will dig into this as we explore this issue.
